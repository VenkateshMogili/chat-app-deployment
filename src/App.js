import React from 'react';
import { Route, Switch } from "react-router";
import "./App.css";
import Contacts from "./components/Contacts";
import Dashboard from "./components/Dashboard";
import Header from "./components/Header";
import { FetchData } from './components/useFetch';
import { ApolloProvider,HttpLink, InMemoryCache } from '@apollo/react-hooks';
import ApolloClient from 'apollo-boost'

function App() {
	// const httpLink = new HttpLink({ uri: '/graphql' })
const client = new ApolloClient({ uri:'http://localhost:4000/graphql', cache: new InMemoryCache()})
	return (
		<div>
			<Header>
				<Switch>
					<Route path="/contacts" component={Contacts} />
					<ApolloProvider client={client} >
					<Route path="/users" component={FetchData} />
          </ApolloProvider>
					<Route path="/settings">
						<h1 className="ml-5 mt-3">Settings</h1>
					</Route>
					<Route path="/" exact component={Dashboard} />
					<Route path="*">
						<div className="ml-5 mt-3">
							<h1>404 Not Found!</h1>
						</div>
					</Route>
				</Switch>
			</Header>
		</div>
	);
}

export default App;
