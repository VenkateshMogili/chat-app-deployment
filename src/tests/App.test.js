import { mount, shallow } from 'enzyme';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';
import renderer from 'react-test-renderer';
import Header from '../components/Header';
import Dashboard from '../components/Dashboard';
import Contacts from '../components/Contacts';
import ContactsClass from '../components/Contacts-Class';

describe('App Component Test Cases', () => {
  // 1.snapshot
  it('Snapshot create/update test case', () => {
    const AppComponent = renderer.create(
      <Router>
        <App />
      </Router>
    ).toJSON();
    expect(AppComponent).toMatchSnapshot();
  })
  // 2.complete component
  it('Snapshot create/update test case for Header', () => {
    const HeaderComponent = mount(
      <Router>
        <Header />
      </Router>
    );
    expect(HeaderComponent).toMatchSnapshot();
  })
  // 3.isolated component
  it('Snapshot create/update test case for Dashboard', () => {
    const HeaderComponent = mount(
      <Router>
        <Dashboard />
      </Router>
    );
    expect(HeaderComponent).toMatchSnapshot();
  })

})


