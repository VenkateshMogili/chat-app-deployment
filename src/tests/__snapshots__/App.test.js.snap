// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test Cases Snapshot create/update test case 1`] = `
<div>
  <div>
    <div
      className="d-flex flex-row"
    >
      <nav
        className="sidebar d-none d-sm-none d-md-block"
      >
        <div
          className="sticky-top h100"
        >
          <ul
            className="nav flex-column"
          >
            <li
              className="nav-item mt-3"
            >
              <div
                className="nav-link"
              >
                <i
                  className="fa fa-bars text-white"
                />
              </div>
            </li>
            <br />
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                <i
                  className="fa fa-home"
                />
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/contacts"
                onClick={[Function]}
              >
                <i
                  className="fa fa-user"
                />
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/settings"
                onClick={[Function]}
              >
                <i
                  className="fa fa-gear"
                />
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <div
        style={
          Object {
            "backgroundColor": "white",
            "color": "black",
            "padding": 20,
          }
        }
      >
        <div>
          <h1>
            Loading...
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Component Test Cases Snapshot create/update test case for Dashboard 1`] = `ReactWrapper {}`;

exports[`App Component Test Cases Snapshot create/update test case for Header 1`] = `ReactWrapper {}`;
